pkgs/tools/text/source-highlight/default.nix

```diff
@@ -9,6 +9,15 @@ stdenv.mkDerivation rec {
     sha256 = "148w47k3zswbxvhg83z38ifi85f9dqcpg7icvvw1cm6bg21x4zrs";
   };
 
+  patches = [
+    # gcc-11 compat upstream patch
+    (fetchpatch {
+      url = "http://git.savannah.gnu.org/cgit/src-highlite.git/patch/?id=904949c9026cb772dc93fbe0947a252ef47127f4";
+      sha256 = "1wnj0jmkmrwjww7qk9dvfxh8h06jdn7mi8v2fvwh95b6x87z5l47";
+      excludes = [ "ChangeLog" ];
+    })
+  ];
+
   # source-highlight uses it's own binary to generate documentation.
   # During cross-compilation, that binary was built for the target
   # platform architecture, so it can't run on the build host.
```

Comment:
Please change this patchPhase to postPatch otherwise your patch is not being applied.

Reply:
also since there is a patchPhase conditionally defined, the `run postPatch` needs to be added in there I think.

Reply:
Oh, great catch on `patchPhase`! Updated the change to use `postPatch`. I think `patchPhase` is unconditional and always calls `postPatch` if that is defined.